.github/workflows/shared.yml

```diff
@@ -49,10 +54,12 @@ jobs:
           version: 0.9.5
 
       - name: Install the project
-        run: uv sync --frozen --all-extras --python ${{ matrix.python-version }} --resolution ${{ matrix.dep-resolution }}
+        run: uv sync ${{ matrix.dep-resolution.install-flags }} --all-extras --python ${{ matrix.python-version }}
 
       - name: Run pytest
-        run: uv run --frozen --no-sync pytest
+        run: uv run ${{ matrix.dep-resolution.install-flags }} --no-sync pytest
+        env:
+          UV_RESOLUTION: ${{ matrix.dep-resolution.name == 'lowest-direct' && 'lowest-direct' || 'highest' }}
```

Comment:
This is needed to ensure subprocesses use the same resolution strategy when they run uv sync rather than overwrite the venv as in @maxisbey's comment: https://github.com/modelcontextprotocol/python-sdk/pull/1325#issuecomment-3242341794